pkg/extension/opampextension/opamp_agent.go

```diff
@@ -309,41 +315,74 @@ func (o *opampAgent) createAgentDescription() error {
 	return nil
 }
 
-func (o *opampAgent) loadEffectiveConfig(path string) error {
-	var k = koanf.New(".")
+func (o *opampAgent) loadEffectiveConfig(dir string) error {
+	if _, err := os.Stat(dir); err != nil {
+		return err
+	}
 
-	rb, err := os.ReadFile(path)
+	ec := map[string]*protobufs.AgentConfigFile{}
 
+	paths, err := filepath.Glob(filepath.Join(dir, "*.yaml"))
 	if err != nil {
 		return err
 	}
 
-	if err := k.Load(rawbytes.Provider(rb), yaml.Parser()); err != nil {
-		return err
-	}
+	for _, p := range paths {
+		var k = koanf.New(".")
 
-	ecb, err := k.Marshal(yaml.Parser())
-	if err != nil {
-		return err
+		rb, err := os.ReadFile(p)
+
+		if err != nil {
+			return err
+		}
+
+		if err := k.Load(rawbytes.Provider(rb), yaml.Parser()); err != nil {
+			return err
+		}
+
+		fb, err := k.Marshal(yaml.Parser())
+		if err != nil {
+			return err
+		}
+
+		ec[filepath.Base(p)] = &protobufs.AgentConfigFile{Body: fb}
 	}
 
-	o.effectiveConfig = string(ecb)
+	o.effectiveConfig = ec
 
 	return nil
 }
 
-func (o *opampAgent) saveEffectiveConfig(path string) error {
-	f, err := os.Create(path)
+func (o *opampAgent) saveEffectiveConfig(dir string) error {
+	d, err := os.Open(dir)
 	if err != nil {
 		return err
 	}
-	defer f.Close()
 
-	_, err = f.Write([]byte(o.effectiveConfig))
+	files, err := d.Readdir(0)
 	if err != nil {
 		return err
 	}
 
+	for _, f := range files {
+		os.Remove(filepath.Join(dir, f.Name()))
```

Comment:
Errors aren't checked here, and in the worst case, this could lead to the wrong configuration being present in the directory with no error reported, if the deleted file isn't present in the list of files to be created.